lua/nvim-treesitter/parsers.lua

```diff
@@ -5,11 +5,12 @@ local M = {}
 
 function M.has_parser(lang)
   local lang = lang or api.nvim_buf_get_option(0, 'filetype')
+  if not lang or #lang == 0 then return false end
```

Comment:
why checking for the string length ?

Reply:
When opening a new buffer with `:enew` it is created with no filetype. Calling `:lua print(vim.api.nvim_buf_get_option(0, "ft")) ` on it will return an empty string instead of `nil`. Then, creating the parsers will cause error. I really think that this case should be handled upstream though, we may want to open a PR for this in the future.

Reply:
i dont see why it would create an error, an empty string should behave as any random string, but if you encountered some issue then i think we should leave it as is.

Reply:
It had created errors actually. Errors while loading the shared library within treesitter. I might look a bit into this soon.

Reply:
i think the `has_parser` function is not working properly because it doesn't look for the `.so` extension but it looks for `.*` so obviously it fails

Reply:
i'm gonna fix this on the current refactoring i'm working on, should be more stable and the api will be available to external plugins too.